adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/uap/PlatformInformation.cs

```diff
@@ -90,33 +141,8 @@ public override string GetDeviceModel()
 
         public override async Task<bool> IsUserLocalAsync(RequestContext requestContext)
         {
-            if (!UserInformation.NameAccessAllowed)
-            {
-                // The access is not allowed and we cannot determine whether this is a local user or not. So, we do NOT add form auth parameter.
-                // This is the case where we can advise customers to add extra query parameter if they want.
-
-                var msg =
-                    "Cannot access user information to determine whether it is a local user or not due to machine's privacy setting.";
-                requestContext.Logger.Info(msg);
-                requestContext.Logger.InfoPii(msg);
-
-                return false;
-            }
-
-            try
-            {
-                return string.IsNullOrEmpty(await UserInformation.GetDomainNameAsync().AsTask().ConfigureAwait(false));
-            }
-            catch (UnauthorizedAccessException)
-            {
-                var msg = "Cannot try Windows Integrated Authentication due to lack of Enterprise capability.";
-                requestContext.Logger.Info(msg);
-                requestContext.Logger.InfoPii(msg);
-
-                // This mostly means Enterprise capability is missing, so WIA cannot be used and
-                // we return true to add form auth parameter in the caller.
-                return true;
-            }            
+            IReadOnlyList<User> users = await User.FindAllAsync();
+            return users.Any(u => u.Type == UserType.LocalGuest || u.Type == UserType.LocalGuest);
```

Comment:
>u.Type == UserType.LocalGuest || u.Type == UserType.LocalGuest) [](start = 34, length = 63)

see comment for MSAL as well

Reply:
Will you update this one as well?

---
In reply to: [206241535](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1153#discussion_r206241535) [](ancestors = 206241535)

Reply:
LocalUser ???

---
In reply to: [206592998](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1153#discussion_r206592998) [](ancestors = 206592998,206241535)